Intro01/Reflex/src/App.rfx

```diff
@@ -0,0 +1,246 @@
+SERIES_AUTHORITY="datacapture";
+BLOB_AUTHORITY="tutorialBlob";
+DOC_AUTHORITY="tutorialDoc";
+
+BLOB_URI="blob://"+BLOB_AUTHORITY;
+DOC_URI="document://"+DOC_AUTHORITY;
+SERIES_URI="series://"+SERIES_AUTHORITY;
+
+CSV_FILE = ENV.RAPTURE_TUTORIAL_CSV;
+if (CSV_FILE == null) do
+  CSV_FILE = "introDataInbound.csv";
+  println("No CSV specified. Defaulting to "+CSV_FILE);
+end
+
+// **********************
+// Create repos if needed
+// **********************
+
+if (!#blob.blobRepoExists(BLOB_URI)) do
+     #blob.createBlobRepo(BLOB_URI, "BLOB {} USING MONGODB {prefix=\""+BLOB_AUTHORITY+"\" }", "REP {} USING MONGODB { prefix=\""+BLOB_AUTHORITY+"\"}");
+end
+if (!#doc.docRepoExists(DOC_URI)) do
+    #doc.createDocRepo(DOC_URI, "REP {} USING MONGODB {prefix=\""+DOC_AUTHORITY+"\"}");
+end
+if (!#series.seriesRepoExists(SERIES_URI)) do
+    #series.createSeriesRepo(SERIES_URI, "SREP {} USING MONGODB {prefix=\""+SERIES_AUTHORITY+"\"}");
+end
+
+// **********************
+// If the data blob does not exist then attempt to load it.
+// Note that the REPL window cannot access the local file system.
+// This will only work when invoked using RaptureRunner. 
+// **********************
+
+// ***
+// ** UPLOAD STEP
+// ***
+
+rawCsvUri = BLOB_URI+"/"+CSV_FILE;
+
+if (!#blob.blobExists(rawCsvUri)) do
+  println("Reading CSV from file " + CSV_FILE);
+  file(CSV_FILE, "CSV") --> rawCsvUri;
+  println("CSV uploaded to "+ rawCsvUri);
+else do
+  println("Data has already been uploaded to the BLOB repository.");
+end
+
+println("\nDone\n");
+
+// **********************
+// Get data from the blob using the pull operation
+// **********************
+
+println("Retrieving raw CSV content from " + rawCsvUri);
+
+// Pull data from the uploaded Blob.
+// Rapture keeps track of the types of blobs.
```

Comment:
Again, better served in the tutorial documentation. (Certainly isn't documented anywhere else)